rules/manager.go

```diff
@@ -465,10 +465,18 @@ func (c ruleDependencyController) AnalyseRules(rules []Rule) {
 	}
 }
 
+// ConcurrentRules represents a slice of indexes of rules that can be evaluated concurrently.
+type ConcurrentRules []int
+
 // RuleConcurrencyController controls concurrency for rules that are safe to be evaluated concurrently.
 // Its purpose is to bound the amount of concurrency in rule evaluations to avoid overwhelming the Prometheus
 // server with additional query load. Concurrency is controlled globally, not on a per-group basis.
 type RuleConcurrencyController interface {
+	// SplitGroupIntoBatches returns an ordered slice of of ConcurrentRules, which are batches of rules that can be evaluated concurrently.
+	// The rules are represented by their index in the original list of rules of the group.
+	// We need the original index to update the group's state and we don't want to mutate the group's rules.
```

Comment:
```suggestion
```

not sure I understand this, would you elaborate?

Reply:
I think removing it is the correct thing to do. It doesn't say anything